bindings/python/src/models.rs

```diff
@@ -173,21 +165,59 @@ impl PyBPE {
     }
 }
 
-/// WordPiece Model
-#[pyclass(extends=PyModel, module = "tokenizers.models", name=WordPiece)]
-pub struct PyWordPiece {}
-
 #[pymethods]
-impl PyWordPiece {
+impl PyBPE {
     #[new]
     #[args(kwargs = "**")]
-    fn new(vocab: Option<&str>, kwargs: Option<&PyDict>) -> PyResult<(Self, PyModel)> {
-        let mut builder = WordPiece::builder();
+    fn new(
+        vocab: Option<&PyAny>,
```

Comment:
I tried it, and got a super weird compilation error that said `Mismatched types`. The error was not caught by `check` nor `clippy` and no source line was mentionned. I supposed it happened in some macro magic that got erased in the compiler warning. I thought it could be left for another PR.

Attempt looked like
```rust
#[derive(PyObject)]
enum PyVocab<'a>{
   Vocab(Vocab),
   Filename(&'a str)
}

// Same error occurs without lifetimes
```

Reply:
Tried to do it again to check error message, it worked this time..... Must have been something else.